src/Form/SettingsForm.php

```diff
@@ -344,6 +344,41 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ];
     }
 
+    $form['tfa_flood'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('TFA Flood Settings'),
+      '#description' => $this->t('Configure the TFA Flood Settings.'),
+    ];
+
+    // Flood control identifier.
+    $form['tfa_flood']['tfa_flood_uid_only'] = [
+      '#type' => 'checkbox',
```

Comment:
The way flood control works is that it stores an identifier in a variable `events`. Each flood control has a name and an identifier and we check for the number of requests made in the last x mins by using the identifier. 
Now identifier can be of two types:
- User id only
- User id and ip address

So when using the userid identifier we check for requests made for that userid but when using the other option we check both the user id and ipaddress.

If a user has been rate limited and the ipaddress is being checked too then he can just change the ipaddress using a proxy and he will be able to do Two Factor auth again.
But in case he we use the uid only option the flood check is independent of the ipaddress and applies on account level.

Here is the relevant code.
```
    if ($this->tfaSettings->get('tfa_flood_uid_only')) {
      // Register flood events based on the uid only, so they apply for any
      // IP address. This is the most secure option.
      $this->floodIdentifier = $values['account']->id();
    }
    else {
      // The default identifier is a combination of uid and IP address. This
      // is less secure but more resistant to denial-of-service attacks that
      // could lock out all users with public user names.
      $this->floodIdentifier = $values['account']->id() . '-' . $this->getRequest()->getClientIP();
    }
```